dags/image_loader.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+import textwrap
+import uuid
+from io import BytesIO
+
+import requests
+from PIL import Image, ImageDraw
+from airflow.exceptions import AirflowException
+from airflow.providers.http.hooks.http import HttpHook
+
+logger = logging.getLogger(__name__)
+
+
+def load_image(**kwargs):
+    http = HttpHook(http_conn_id="image_api", method="GET")
+
+    try:
+        response = http.run(endpoint="")
+        response.raise_for_status()
+        image_url = response.url
+        logger.info("Successfully loaded image URL.")
+        kwargs["ti"].xcom_push(key="image_url", value=image_url)
+    except requests.exceptions.HTTPError as e:
+        logger.error(f"HTTP error: {e}")
+        raise AirflowException(f"Failed to load image: {e}")
+    except requests.exceptions.RequestException as e:
+        logger.error(f"Request exception: {e}")
+        raise AirflowException(f"Failed to load image: {e}")
+
+    response = requests.get(image_url)
+    if response.status_code != 200:
+        raise AirflowException(f"Failed to download image: {response.status_code}")
+
+    image = Image.open(BytesIO(response.content))
+
+    quote = kwargs["ti"].xcom_pull(task_ids="load_quote", key="quote")
+    author = kwargs["ti"].xcom_pull(task_ids="load_quote", key="author")
+
+    draw = ImageDraw.Draw(image)
```

Comment:
It's a bit weird decisions to combine text and photo here. According to single responsibility principle, this function should just send data to MS teams - and in your variant you are also trying to apply some transformations to it. Try to move this to another step